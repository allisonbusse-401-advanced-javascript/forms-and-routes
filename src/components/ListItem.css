.ListItem {
  list-style: none;
  background-color: lightsalmon;
  margin-right: 15px;
  padding-bottom: 10px;
}
    a:visited {
      color: darkslategray;
    }

    a {
      color: darkslategray;
    }
    p {
      text-align: center;
    }