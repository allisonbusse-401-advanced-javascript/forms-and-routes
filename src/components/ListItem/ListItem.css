.ListItem {
  list-style: none;
  background-color: lightsalmon;
  margin-right: 15px;
  padding-bottom: 10px;
}
    a:visited {
      color: darkslategray;
    }

    a {
      color: darkslategray;
    }
    p {
      text-align: center;
      font-family: Arial, Helvetica, sans-serif;
    }