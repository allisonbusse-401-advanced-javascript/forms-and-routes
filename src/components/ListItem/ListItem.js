import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './ListItem.css';
import { Link } from 'react-router-dom';


class ListItem extends Component {

  static propTypes = {
    _id: PropTypes.string,
    name: PropTypes.string,
    photo: PropTypes.string,
    history: PropTypes.object
  }


  render() {
    const characterUrl = `/character/${this.props._id}`;
    return (
      <Link to={characterUrl}>

        <li className={styles.ListItem} key={this.props._id}>
          <img src={this.props.photo} alt={this.props._id}></img>
          <p>{this.props.name}</p>
        </li>
      </Link>
    );
  }
}

export default ListItem;


// export default function ListItem({ key, name, photo, history }) {
//   return (
//     <li key={key} onClick={handleClick}>
//       <img src={photo}></img>
//       <p>{name}</p>
//     </li>
//   );
// }

// function handleClick(event) {
//   event.preventDefault();
//   this.props.history.push(`/character/${name}`);
// }



// ListItem.propTypes = {
//   key: PropTypes.string,
//   name: PropTypes.string,
//   photo: PropTypes.string,
//   history: PropTypes.object
// }
//   ;
