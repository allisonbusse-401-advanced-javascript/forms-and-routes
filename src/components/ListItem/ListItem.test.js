import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';

describe('ListItem component', () => {
  it('renders ListItem', () => {
    const item = {
      _id: '1234',
      name: 'allison',
      photoUrl: 'hey'
    };
    const wrapper = shallow(<ListItem
      {...item} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
