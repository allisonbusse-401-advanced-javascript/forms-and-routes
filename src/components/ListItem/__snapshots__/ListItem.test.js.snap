// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders ListItem 1`] = `
<Link
  to="/character/1234"
>
  <li
    className="ListItem"
    key="1234"
  >
    <img
      alt="1234"
    />
    <p>
      allison
    </p>
  </li>
</Link>
`;
